emmet-core/emmet/core/qchem/calculation.py

```diff
@@ -0,0 +1,669 @@
+    @classmethod
+    def validate_cds_gradients(cls, v):
+        if v is not None and not isinstance(v, (np.ndarray, Dict, List)):
+            raise ValueError(
+                "cds_gradients must be a numpy array, a dict or a list or None."
+            )
+        return v
+
+    dipoles: Optional[Dict[str, Any]] = Field(
+        None, description="The associated dipoles for Mulliken/RESP/ESP charges"
+    )
+
+    gap_info: Optional[Dict[str, Any]] = Field(
+        None, description="The Kohn-Sham HOMO-LUMO gaps and associated Eigenvalues"
+    )
+
+    @classmethod
+    def from_qcoutput(cls, qcoutput: QCOutput) -> "CalculationOutput":
+        """
+        Create a QChem output document from a QCOutput object.
+
+        Parameters
+        ----------
+        qcoutput
+            A QCOutput object.
+
+        Returns
+        --------
+        CalculationOutput
+            The output document.
+        """
+
+        return cls(
+            optimized_molecule=qcoutput.data.get(
+                "molecule_from_optimized_geometry"
+            ).as_dict()
+            if qcoutput.data.get("molecule_from_optimized_geometry")
+            else {},
+            mulliken=qcoutput.data.get(["Mulliken"][-1], []),
+            esp=qcoutput.data.get(["ESP"][-1], []),
+            resp=qcoutput.data.get(["RESP"][-1], []),
+            nbo_data=qcoutput.data.get("nbo_data", {}),
+            frequencies=qcoutput.data.get("frequencies", {}),
+            frequency_modes=qcoutput.data.get("frequency_mode_vectors", []),
+            final_energy=qcoutput.data.get("final_energy", None),
+            enthalpy=qcoutput.data.get("enthalpy", None),
+            entropy=qcoutput.data.get("entropy", None),
+            scan_energies=qcoutput.data.get("scan_energies", {}),
+            scan_geometries=qcoutput.data.get("optimized_geometries", {}),
+            scan_molecules=qcoutput.data.get("molecules_from_optimized_geometries", {}),
+            pcm_gradients=qcoutput.data.get(["pcm_gradients"][0], None),
+            cds_gradients=qcoutput.data.get(["CDS_gradients"][0], None),
+            dipoles=qcoutput.data.get("dipoles", None),
+            gap_info=qcoutput.data.get("gap_info", None),
+        )
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+    # TODO What can be done for the trajectories, also how will walltime and cputime be reconciled
+
+
+class Calculation(BaseModel):
+    """Full QChem calculation inputs and outputs."""
+
+    dir_name: str = Field(None, description="The directory for this QChem calculation")
+    qchem_version: str = Field(
+        None, description="QChem version used to perform the current calculation"
+    )
+    has_qchem_completed: Union[QChemStatus, bool] = Field(
+        None, description="Whether QChem calculated the calculation successfully"
+    )
+    input: CalculationInput = Field(
+        None, description="QChem input settings for the calculation"
+    )
```

Comment:
Is this meant to be the input document? If so, please clarify the docstsring for consistency with `output`, which says explicitly "calculation output document`